Wearables - Rebuild

awatson1978 opened the pull request:
Rebuild of PR https://github.com/synthetichealth/synthea/pull/1254

dehall commented on config/simulations/README.md:
```diff
@@ -9,7 +9,76 @@ files will be in the `output/physiology` directory.
+
+## Configuration  
+
+You will need to set the following two fields to `true` in the `src/main/resources/synthea.properties` file.  
```
There are two different ways the physiology models can be invoked and this is conflating them a bit.  (1) The `./gradlew physiology --args="config/simulations/[config name].yml"` command right below this invokes a simulation defined in that file, those simulations point to one of the physiology models. (2) You can also create a module with a Physiology state, run Synthea with it, and when that state gets hit it will run the corresponding model. Only (2) here requires these two settings.

awatson1978 replied:
Removed.

dehall reviewed (commented):
The details of the new simulations and models are a little bit over my head so I can't comment too much on those. In general they look fine to me. The 4 new modules though, I'm not super clear on the intent of those and I think that needs to be clear before we decide where to go with this. 3 of the 4 new modules are named "wearables" but wearable devices are only a piece of what's in the module. All of them assign a new Condition to an individual which I see is meant to be the reason an individual uses a wearable but I would want to see more detail in the subsequent clinical progression of the modules. Are they intended to be top-level modules? (ie, are these meant to be run for the entire population?)

awatson1978 commented on src/main/resources/modules/menstrual_cycle.json:
```diff
@@ -0,0 +1,276 @@
+      "symptom": "Cramps",
+      "cause": "Endometriosis",
+      "direct_transition": "Menstrual_Cramps_Symptom_Ends",
+      "codes": [
```
Hrmmm.  There were a few non-terminated paths in earlier versions of the modules (~July), although all of the new modules currently compile in the Module Builder for me.  I'm not able to replicate the builder crash based on the `codes` field.  :/

awatson1978 replied:
Removed `codes` field from Symptom states.

awatson1978 replied:
I overhauled the entire module, and it's now simply an Endometriosis module.  I went through and rebuilt most all of the states, using the editor.  Turns out the Symptom didn't have a probability %, which was what was causing it to crash.   Unit tests running good.

awatson1978 commented on src/main/resources/modules/menstrual_cycle.json:
```diff
@@ -0,0 +1,276 @@
+      "lead_time": 0,
+      "inputs": [
+        {
+          "from_exp":  "#{BMI}",
```
`mass` is the internal representation in the physiology model, on page 10 of Roblitz2013_Menstrual_Cycle.pdf. 

<img width="520" alt="Screenshot 2023-10-02 at 9 05 04 PM" src="https://github.com/synthetichealth/synthea/assets/675910/98c80eac-7367-44cd-b53f-7c38bca8955c">

awatson1978 replied:
I'm not entirely sure the assignment is really doing anything though.  The `mass` parameter is used in a Kinetic Law and a Rule, and I think it's getting re-calculated by other parameters in the equation.  

<img width="482" alt="Screenshot 2023-10-02 at 9 08 54 PM" src="https://github.com/synthetichealth/synthea/assets/675910/357e277d-ac19-4a93-a4d9-222302744ff6">

<img width="495" alt="Screenshot 2023-10-02 at 9 09 11 PM" src="https://github.com/synthetichealth/synthea/assets/675910/4db65e6a-56ca-4acb-bd8a-b12a8dc8f7c4">

awatson1978 replied:
I'm thinking it might be better to try not to feed BMI into the equation at all, and to just feed in a static constant instead.

awatson1978 commented on src/main/resources/modules/wearables_circadian.json:
```diff
@@ -0,0 +1,347 @@
+        "display": "Monitor, device (physical object)"
+      },
+      "direct_transition": "SleepStudy_Baseline",
+      "assign_to_attribute": "snf_iwatch"
```
`snf_iwatch` renamed to `sleep_data_iwatch`, `ecg_data_iwatch`, `o2_data_iwatch`, and `o2_data_cpap`.  Unique data assignment variable names for each module.

awatson1978 commented on src/main/resources/modules/menstrual_cycle.json:
```diff
@@ -0,0 +1,276 @@
+      "remarks": [
+        "About one-in-five Americans use a smart watch or fitness tracker.  \n https://www.pewresearch.org/short-reads/2020/01/09/about-one-in-five-americans-use-a-smart-watch-or-fitness-tracker/"
+      ],
+      "distributed_transition": [
```
All distributions now adding up to 100%.  Still working on making sure only the correct % in the overall population get the symptom.

dehall commented on src/main/resources/modules/menstrual_cycle.json:
```diff
@@ -0,0 +1,350 @@
+    },
+
+    "Endometriosis_Risk": {
+      "type": "ConditionOnset",
```
These "Risk" states either need to have `codes` defined, or they need to be a different state type. As it stands they cause exceptions when running, see the unit tests: https://github.com/synthetichealth/synthea/actions/runs/6397186772/job/17364591481

If you don't intend for any record artifact to be added for this, you probably just want these to be Simple states

awatson1978 replied:
Completely overhauled the entire module.  Refactored from `Endometriosis_Risk` to simply `Endometriosis`.   Has a matching ConditionEnd.   Unit tests passing.   No crashing recently.

jawalonoski commented:
The wearables sleep apnea module needs a good scrubbing.

In the image below, you can see a few potential issues:

- The `Wellness_Baseline_Encounter` is an `emergency` encounter, which _might be fine_, but then it probably isn't a "wellness" encounter. And if it is, then maybe click "Add Wellness".
- The `Wellness_Baseline_Encounter` transitions directly to the `SleepStudy_Baseline_Encounter`. The module engine will halt at this point, because it is impossible for there to be two simultaneous encounters. The first encounter _shall_ end before the second one begins.
- The `SleepStudy_Baseline_Encounter` is an `emergency` encounter, which also seems incorrect.

<img width="1085" alt="image" src="https://github.com/synthetichealth/synthea/assets/1054765/a6139967-707e-4652-9268-bb48f4863037">

The `Sleep_Apnea_Resolved` state is not correct. You can tell, because if you click on it, the module builder does not actually reference a condition. Click "Add Condition Onset" (or "Add Codes") if you prefer.

<img width="785" alt="image" src="https://github.com/synthetichealth/synthea/assets/1054765/3b78a97b-a84f-47bd-bdf1-cb912e497686">

Similarly, some of the `Observation` states may be incomplete and could export into invalid FHIR. They have no units of measure, and they have no values, for instance `SleepAnaysis` state outputs `Sleep duration.`

dehall commented on src/main/resources/modules/wearables_ecg.json:
```diff
@@ -0,0 +1,603 @@
+      "remarks": [
+        "About one-in-five Americans use a smart watch or fitness tracker.  \n https://www.pewresearch.org/short-reads/2020/01/09/about-one-in-five-americans-use-a-smart-watch-or-fitness-tracker/"
+      ],
+      "distributed_transition": [
```
These transitions as implemented result in everyone getting the "Sedentary Job" condition eventually, which doesn't seem like what we want. Should this be more like the endometriosis module which has some % go directly to terminal?

awatson1978 replied:
Updated logic:

All adults → Age_18_65 (everyone reaches adulthood)
Age_18_65 → 30% adopt wearables, 70% go to Terminal
Wearable_Device_Adoption → 60% fitness-only, 40% clinical monitoring

New Population Distribution:

70% - No wearables at all
18% - Fitness tracking only (30% × 60%)
12% - Clinical monitoring with wearables (30% × 40%)

dehall commented on src/main/resources/modules/wearables_ecg.json:
```diff
@@ -0,0 +1,603 @@
+      ],
+      "name": "Sedentary_Lifestyle"
+    },
+    "Symptom_Onset_Delay": {
```
The symptoms that get added in this sequence are immediately ended right after, so they don't really do anything. Then the ConditionEnds after that aren't going to work because they point to a Guard instead of a ConditionOnset. I'd suggest just having this whole branch (the "80%" after the guard) go directly to Terminal

dehall reviewed (commented):
Ok I think this is really close but still a few things that should be cleaned up before we merge. I recognize these new modules are meant to be quick demos that get the physiology content into records, rather than comprehensive modules of the conditions in question, but since they'll be part of every synthea population going forward I do want to make sure what these new modules create is as good as we can make it.

awatson1978 commented:
**Wearables ECG Module (wearables_cardiac_monitoring.json)**

✅ Fixed Critical Issues:
Renamed module to "Wearables_Cardiac_Monitoring" for clarity
Updated adoption rate from 20% to 30% based on 2024 statistics
Simplified population model - removed complex 9-age-group stratification
Fixed encounter classes - changed from "emergency" to "ambulatory"
Updated documentation with current research citations
Streamlined age targeting to 18-65 years (prime wearable adoption demographic)

**Sleep Apnea Module (wearables_sleepapnea.json)**

✅ Fixed Critical Logic Error:

Removed problematic Guard condition that prevented sleep apnea onset modeling
Added proper condition onset flow with Sleep_Apnea_Risk_Assessment → Sleep_Apnea_Onset
Added missing probability fields to symptom states
Fixed encounter classes from "emergency" to "ambulatory"
Added age-based risk stratification (≥30 years, 5% risk)

✅ Key Improvements Made:

Runtime Stability - All modules now execute without crashes
Clinical Accuracy - Updated statistics reflect 2024 research
Proper Medical Coding - Fixed medication and encounter coding issues
Realistic Population Modeling - Simplified and updated adoption rates
Logical Flow - Fixed state transition logic throughout

awatson1978 commented:
**Wearables - Cardiac Monitoring**  
<img width="532" alt="Screenshot 2025-06-07 at 11 20 23 PM" src="https://github.com/user-attachments/assets/7e7fa4f7-8ac1-4a2f-a168-29370f3b2b68" />

**Wearables - Sleep Apnea**  
<img width="855" alt="Screenshot 2025-06-07 at 11 19 48 PM" src="https://github.com/user-attachments/assets/60b8e9c0-3551-4bbe-952a-e4f9b327be80" />

**Endometriosis**  
<img width="619" alt="Screenshot 2025-06-07 at 11 19 13 PM" src="https://github.com/user-attachments/assets/5d781af7-ef1e-4cd5-bb09-4d657c3850d8" />

awatson1978 commented:
Take a look at the updated changes?  
@jawalonoski @hadleynet
